include/swift/SILOptimizer/Utils/ConstExpr.h

```diff
@@ -40,6 +43,106 @@ class SymbolicValueAllocator;
 class ConstExprFunctionState;
 class UnknownReason;
 
+/// This class is a utility used for evaluating string literals.
+/// It retrieves info about the string and allows conversion
+/// from a swift string literal to a string ref. The class is designed
+/// to be used in conjunction with SILInstructions or SILValues.
+class StringLiteralInfo {
+  StringLiteralInfo() = default;
+
+public:
+  bool isAscii;
+  StringRef value;
+
+  /// Create an instance of StringLiteralInfo by casting v to an ApplyInst.
+  /// \param v SILValue to be cast to ApplyInst and used in creation of the string literal.
```

Comment:
One ask. Can you put a newline in between "Create an instance" and "\param v". Can you do something similar in the rest of the changes?